api/maven-api-core/src/main/java/org/apache/maven/api/Session.java

```diff
@@ -73,15 +74,12 @@ public interface Session {
     Map<String, String> getSystemProperties();
 
     /**
-     * Gets the immutable effective properties to use for interpolation. The effective properties are collected from
-     * {@link #getSystemProperties()} and {@link #getUserProperties()} in this order, and more.
-     * <p>
-     * These properties are "effective" in a way they are built obeying all "user overrides" and others.
+     * Gets the effective properties to use for interpolation.
      *
-     * @return the config properties, never {@code null}
+     * @return the effective properties, never {@code null}
      */
     @Nonnull
-    Map<String, Object> getEffectiveProperties();
+    Map<String, String> getEffectiveProperties(@Nullable Project project);
```

Comment:
We should specify if this map is mutable or not, and if mutable, if a new map is created each time.

Reply:
Just did that with last two "javadoc" commit: ... each call returns "new copy" of ...